xcode/Subconscious/Shared/Components/Common/Tabs/TabButtonView.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  TabButtonView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 28/3/2023.
+//
+
+import SwiftUI
+
+struct TabButtonView<Label: View>: View {
+    var action: () -> Void
+    let label: Label
+    var selected: Bool
+
+    init(@ViewBuilder label: () -> Label) {
+        self.label = label()
+        self.selected = false
+        self.action = {}
+    }
+    
+    init(@ViewBuilder label: () -> Label, action: @escaping () -> Void, selected: Bool) {
+        self.label = label()
+        self.action = action
+        self.selected = selected
+    }
```

Comment:
I could drop this, it's required for icon support in tabs but if we stick to plain text we don't need to worry about `@ViewBuilder`.